### PR TITLE
KOGITO-5413 - Upgrade Java release version to v11

### PR DESCRIPTION
**Pull request by cristianonicolai**
Assembly:
- https://github.com/kiegroup/kogito-runtimes/pull/1401
- https://github.com/kiegroup/kogito-apps/pull/906
- https://github.com/kiegroup/kogito-examples/pull/747

### CODE REVIEW

**Comment by cristianonicolai** [comment]
Jenkins rerun runtimes tests

**Comment by mareknovotny** [comment]
I am ok with this, but as Quarkus did, we should very probably change the kogito major version release for this too so instead of targeting it for 1.9.0.Final, skip that version and do 2.0.0.Final.

**Comment by danielezonca** [comment]
> I am ok with this, but as Quarkus did, we should very probably change the kogito major version release for this too so instead of targeting it for 1.9.0.Final, skip that version and do 2.0.0.Final.

Personally I would prefer to do it now for multiple reasons:
- Quarkus 2.0 is already jdk 11 only so keeping bytecode compatibility with 8 is just an internal limitation
- I don't think Kogito 2.0 will be released just after 1.9, personally I think we will need a bit more time
- JDK8 compatibility in Kogito is just an internal detail: we always report in doc and examples JDK 11 as required

**Review by danielezonca — APPROVED** [review]

**Comment by mareknovotny** [comment]
you can release now 2.0, just because of that change.The thing is you are  now changing the release and target byte code of all built jars too

**Review by tiagodolphine — APPROVED** [review]

**Review by ricardozanini — APPROVED** [review]
:palms_up_together:

File: kogito-build/kogito-build-parent/pom.xml [hunk 1]
```diff
@@ -51,7 +51,9 @@
     <container.image.kafka>confluentinc/cp-kafka:${version.org.confluent.kafka}</container.image.kafka>
     <container.image.mongodb>mongo:${version.org.mongo}</container.image.mongodb>
 
-    <maven.compiler.release>8</maven.compiler.release>
+    <maven.compiler.release>11</maven.compiler.release>
+    <maven.compiler.source>11</maven.compiler.source>
+    <maven.compiler.target>11</maven.compiler.target>
```
Comment (MarianMacik) [inline kogito-build/kogito-build-parent/pom.xml:56]: Isn't the release property enough?
See https://stackoverflow.com/questions/43102787/what-is-the-release-flag-in-the-java-9-compiler

**Review by MarianMacik — APPROVED** [review]
Looks good, I have just 2 things:
* Isn't the release property enough?
* What was the purpose of our "internal" limitation to JDK 8 when we anyway require JDK 11 by the enforcer?

**Comment by krisv** [comment]
Imho we allowed Java 1.8 for technical reasons (because Quarkus required this for the extensions) but we have always recommended a JDK 11 minimum to users / customers.  Therefore I think we should just do this without bumping the major version, as there are other things we are planning to be part of the Kogito 2.0 but these are not feasible in a short time frame, and I don't think doing another major update soon after would be recommended.

File: kogito-build/kogito-build-parent/pom.xml (hunk 1 above)
Comment (mareknovotny) [reply kogito-build/kogito-build-parent/pom.xml:56]: no, the defaults for source and target  is 1.6. see https://maven.apache.org/plugins/maven-compiler-plugin/compile-mojo.html

Comment (MarianMacik) [reply kogito-build/kogito-build-parent/pom.xml:56]: Yes, but release will override both, right? From http://openjdk.java.net/jeps/247:
```
--release N is roughly equivalent to:

    for N < 9: -source N -target N -bootclasspath <documented-APIs-from-N>,
    for N >= 9: -source N -target N --system <documented-APIs-from-N>.
```
But who knows :)

**Review by evacchi — APPROVED** [review]

**Comment by ricardozanini** [comment]
@mareknovotny are you ok with this PR?

**Pull request merged**